src/otx/core/model/module/instance_segmentation.py

```diff
@@ -53,10 +53,12 @@ def __init__(
             param_dict = {}
             for name, param in sig.parameters.items():
                 param_dict[name] = param.default
-            param_dict.pop("kwargs")
+            param_dict.pop("kwargs", {})
 
             metric = metric(**param_dict)  # type: ignore[call-arg]
         self.metric = metric
+        if self.metric is not None:
+            self.metric.num_classes = otx_model.num_classes  # type: ignore[attr-defined]
```

Comment:
Move `num_classes` assignment to the metric constructor and construct it at `on_validation_start()` and `on_test_start()`.